src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -607,13 +607,23 @@ private PollingResult compareRemoteRevisionWithImpl(Job<?, ?> project, Launcher
             listener.getLogger().println("[poll] Last Built Revision: " + buildData.lastBuild.revision);
         }
 
-        final EnvVars pollEnv = project instanceof AbstractProject ? GitUtils.getPollEnvironment((AbstractProject) project, workspace, launcher, listener, false) : lastBuild.getEnvironment(listener);
+        EnvVars pollEnv;
+        if (project instanceof AbstractProject) {
+            pollEnv = GitUtils.getPollEnvironment((AbstractProject) project, workspace, launcher, listener, false);
+        } else {
+            try {
+                pollEnv = lastBuild.getEnvironment(listener);
```

Comment:
The original code is probably not ideal, I agree, but wouldn't it be better to wait to approach that particular logic until we have time to do a much larger rewrite of the environment handling?